ts4.1/index.d.ts

```diff
@@ -97,6 +99,15 @@ declare module 'i18next' {
   }
 }
 
+type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
+  ? Record<string, unknown>
+  : never;
+declare module 'react' {
+  interface HTMLAttributes {
+    children?: ReactNode | ObjectOrNever;
+  }
+}
```

Comment:
this override is breaking all components that use render props when upgrading to React 18. We can patch this away, but I would strongly encourage NOT overriding generic types for all components. Can't it be limited to just the `Trans` component? A sledgehammer can't be the only solution when a scalpel will do?

Reply:
Feel free to provide a PR with a better solution that works.

// @marcalexiei do you know about this?

Reply:
@cellog **extending** is different from **overriding**. Share a minimal reproducible example so we can see how we can help you.